packages/theme/src/cli/services/dev.ts

```diff
@@ -227,6 +225,27 @@ export function showDeprecationWarnings(args: string[]) {
       ],
     })
   }
+
+  const legacyFlagPresent = args.find((arg) => arg === '--legacy')
+
+  if (legacyFlagPresent) {
+    renderWarning({
+      headline: ['`SHOPIFY_FLAG_LEGACY` is deprecated.'],
+      body: [
+        'The',
+        {command: 'SHOPIFY_FLAG_LEGACY'},
+        'flag has been deprecated. If this variable is essential to your workflow, please report an issue at',
+        {
+          link: {
+            url: 'https://github.com/Shopify/cli/issues',
+          },
+        },
+        {
+          char: '.',
+        },
+      ],
+    })
+  }
```

Comment:
Because this flag has always been hidden and I've been referring to it as --legacy, I believe we could refer to it as legacy here as well:

```suggestion
  if (legacyFlagPresent) {
    renderWarning({
      headline: ['The', {command: '--legacy'}, 'is deprecated.'],
      body: [
        'The',
        {command: '--legacy'},
        'flag has been deprecated. If this variable is essential to your workflow, please report an issue at',
        {
          link: {
            url: 'https://github.com/Shopify/cli/issues',
          },
        },
        {
          char: '.',
        },
      ],
    })
  }
```

Reply:
<img width="410" alt="image" src="https://github.com/user-attachments/assets/2db5c555-919b-46a2-af09-166bcd91e9f2">